src/ImageSharp/Formats/Pbm/PbmDecoderCore.cs

```diff
@@ -144,14 +145,22 @@ private void ProcessHeader(BufferedReadStream stream)
                 throw new InvalidImageContentException("Unknown of not implemented image type encountered.");
         }
 
-        stream.SkipWhitespaceAndComments();
-        int width = stream.ReadDecimal();
-        stream.SkipWhitespaceAndComments();
-        int height = stream.ReadDecimal();
-        stream.SkipWhitespaceAndComments();
+        if (!stream.SkipWhitespaceAndComments() ||
```

Comment:
I really like this!

Reply:
Only thing I would suggest is to change the method names to use `Try` prefix to match convention.

Reply:
I've pushed those changes.

Reply:
Ha, didn't realize you'd explicitly removed them! I can revert at a later time if you feel strongly about it.

Reply:
I really prefer it to be the other way.

By convention, `TryDoSomething(...) == false` should indicate that the operation failed as a whole. This is not the case with `TryReadDecimal(out value) == false` when it happens at the very las digit in the file: the decoding of the digit was succesful (we should threat the result as valid!) just the file reached an EOF.

I like following such conventions to the letter, because ambiguity can lead to a misunderstandings and programming mistakes.

Reply:
I’m not precious about it (and happy to revert) but there’s something a little clunky about the method there. We’re really returning two things, the decimal and the stream state. I’ve seen something in the runtime I’m sure that handles it better (Maybe inside guid parsing code)

Reply:
I considered returning a `(bool, int)` tuple, but that would lead to much more complicated code on the callsites, since we couldn't do `!stream.SkipWhitespaceAndComments() || ...`. Defining the names without the `Try` prefix + documenting the behavior was the best idea I was able to come up with.